src/algokit/cli/generate.py

```diff
@@ -0,0 +1,124 @@
+            "The TypeScript generator requires Node.js and npx to be installed. Please install Node.js to continue."
+        ) from e
+
+
+def generate_clients(app_spec: pathlib.Path, output: pathlib.Path) -> None:
+    if app_spec.is_dir():
+        for child in app_spec.iterdir():
+            if child.is_dir():
+                generate_clients(child, output)
+            elif child.name.lower() == "application.json":
+                formatted_output = format_client_name(output=output, application_file=child)
+                logger.info(
+                    f"Generating Python client code for application specified in "
+                    f"{child} and writing to {formatted_output}"
+                )
+                algokit_client_generator.generate_client(child, pathlib.Path(formatted_output))
+    else:
+        formatted_output = format_client_name(output=output, application_file=app_spec)
+        logger.info(
+            f"Generating Python client code for application specified in {app_spec} and writing to {formatted_output}"
+        )
+        algokit_client_generator.generate_client(app_spec, pathlib.Path(formatted_output))
+
+
+@click.group("generate", short_help="Generate code for an AlgoKit application.")
+def generate_group() -> None:
+    pass
+
+
+@generate_group.command("client")
+@click.option(
+    "app_spec",
+    "--appspec",
+    "-a",
+    type=click.Path(exists=True, dir_okay=True, resolve_path=True),
+    default="./application.json",
+    help="Path to the application specification file",
+)
+@click.option(
+    "output",
+    "--output",
+    "-o",
+    type=click.Path(exists=False, dir_okay=False, resolve_path=True),
+    default="./client_generated.py",
+    help="Path to the output file",
+)
+@click.option(
+    "--language",
+    default=None,
+    type=click.Choice(["python", "typescript"]),
+    help="Programming language of the generated client code",
+)
+def generate_client(app_spec: str, output: str, language: str | None) -> None:
+    """
+    Generate a client.
+    """
+    output_path = pathlib.Path(output)
+    app_spec_path = pathlib.Path(app_spec)
+    if language is None:
+        extension = output_path.suffix
+        if extension == ".ts":
+            language = "typescript"
+        elif extension == ".py":
+            language = "python"
+        else:
+            raise click.ClickException(
+                f"Unsupported file extension: {extension}. Please use '.py' for Python or .ts' for TypeScript."
+            )
+
+    if language.lower() == "typescript":
+        check_node_installed()
+        proc.run(
+            [
+                "npx",
+                "--yes",
+                "@algorandfoundation/algokit-client-generator@v2.0.0-beta.1",
+                "generate",
+                "-a",
+                app_spec,
+                "-o",
+                output,
+            ],
+            bad_return_code_error_message="npx failed, please check your npm",
```

Comment:
Can we change this to the following? cmd should be set to the command array passed to proc.run

f"Failed to run `{' '.join(cmd)}` for {app_spec}. Is npx installed and available on PATH?"